packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1042,11 +1042,28 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   /// The given [BuildContext] will be rebuilt if the state of the route changes
   /// while it is visible (specifically, if [isCurrent] or [canPop] change value).
   @optionalTypeArgs
-  static ModalRoute<T>? of<T extends Object?>(BuildContext context) {
-    final _ModalScopeStatus? widget = context.dependOnInheritedWidgetOfExactType<_ModalScopeStatus>();
+  static ModalRoute<T>? of<T extends Object?>(
+    BuildContext context, {
+    bool createDependency = true,
```

Comment:
What problem is it solving to not create the dependency? In my experience, this kind of API often ends up being a footgun where people forget to switch this to true when their code evolves and they do actually need to rebuild when ModalRoute changes.

Reply:
In many cases, people obtain route arguments through `ModalRoute.of(context)?.settings.arguments`. In this scenario, there is no need to create dependency. 

Now after upgrading to 3.19, I found that I obtained the  route arguments in Page2 and then pushed to Page3. During the process of Page3 being pushed and popped, the `build` method of Page2 will be called again because of `ModalRoute.of`. In this process, it is meaningless for Page2 to re-obtain routing parameters.

https://github.com/flutter/flutter/assets/19367531/d3514e1b-d051-43ed-84b8-d1092810ec8d

Reply:
> In many cases, people obtain route arguments through ModalRoute.of(context)?.settings.arguments. In this scenario, there is no need to create dependency.

The settings associated with a route (and with that the arguments) can change, though (see [1]). Not creating a dependency will lead to bugs because widgets that previously read the args will not rebuild with the new args. So, this API will likely be a foot gun. Instead, we could consider an API that only rebuilds when the arguments change. Something like `ModalRoute.argumentsOf(context)`.


[1] https://github.com/flutter/flutter/blob/ac5be2d821a0ff4a0c1c8a00f9c7f27a00a894ea/packages/flutter/lib/src/widgets/navigator.dart#L191

Reply:
Good suggestion, but sorry I don’t have time to complete it, thanks for reviewing.